doc/cascadia/Cascading-Default-Settings.md

```diff
@@ -0,0 +1,399 @@
+---
+author: Mike Griese @zadjii-msft
+created on: 2019-05-31
+last updated: 2019-06-13
+issue id: #754
+---
+
+# Cascading Default + User Settings
+
+## Abstract
+
+This spec outlines adding support for a cascading settings model. In this model,
+there are two settings files, instead of one.
+
+1. The default settings file
+2. The user's settings file
+
+The default settings file would be a static, read-only file shipped with the
+terminal. The user settings file would then contain all the user's chosen
+customizations to the settings. These two files would then be composed together
+when the app is launched, so that the runtime settings are the union of both the
+defaults and whatever modifications the user has chosen. This will enable the
+app to always use a default schema that it knows will be valid, and minimize the
+settings that the user needs to customize.
+
+Should the settings schema ever change, the defaults file will change, without
+needing to re-write the user's settings file.
+
+
+## Inspiration
+
+Largely inspired by the settings model that both VS Code (and Sublime Text) use.
+
+### Goal: Minimize Writing to `profiles.json`

```

Comment:
Is this a primary goal? Should we list more goals? If we include dynamic settings, a goal would be to "reduce the requirement for a user to reset their settings to get new defaults"

Reply:
Additional goal: reduce the incidence of application failure destroying user customization; allow the user more settings mobility./

Reply:
Potential goal: This model immediately opens us up to supporting _multiple settings files_. I love the idea of having my color schemes in one file and my profiles piecemeal in another. is that possible? Supportable? A `non-goal` for this spec?

Reply:
Oh, you know, I think I wrote that section before I had much of the rest of the spec written. 

I agree on the dynamic profiles bit.

I don't think I understand the failure case you're describing...

I'll add a note on multiple files to the "Future considerations" section